spmd/tensor/ops/tp_sharding_ops.py

```diff
@@ -4,40 +4,36 @@
 import torch.utils._pytree as pytree
 from typing import List
 from spmd.tensor.api import DTensor
+from spmd.tensor.dispatch import OpSchema, OutputSharding
 from spmd.tensor.utils import unwrap_local_tensor
-from spmd.tensor.ops.utils import unwrap_single_placement, register_impl
+from spmd.tensor.ops.utils import unwrap_single_placement, register_impl, register_prop_rule
+from spmd.tensor.placement_types import DTensorSpec
 
 """
 The ops below were quickly hacked and needed to be polished down the road.
 Although they come with unit tests already, the logic is directly borrowed
-from ShardedTensor. We need to also make it work for all placement types 
+from ShardedTensor. We need to also make it work for all placement types
 of DTensor and all corner cases for sharded distributed tensor.
 """
 
 
-@register_impl("aten.baddbmm.default")
-def dist_baddbmm(
-    self: DTensor,
-    batch1: DTensor,
-    batch2: DTensor,
-    beta: float = 1.0,
-    alpha: float = 1.0,
-) -> DTensor:
-    local_input, local_batch1, local_batch2 = pytree.tree_map(
-        unwrap_local_tensor, (self, batch1, batch2)
+@register_prop_rule("aten.bmm.default")
```

Comment:
Can we move this op to maybe matrix_ops? So that this file only keeps the legacy code. So the ultimate goal is to remove this file completely.